karapace/kafka_rest_apis/__init__.py

```diff
@@ -279,24 +279,31 @@ def get_topic_config(self, topic: str) -> dict:
             return self.admin_client.get_topic_config(topic)
 
     def cluster_metadata(self, topics: Optional[List[str]] = None) -> dict:
```

Comment:
IMO we should avoid writing code that rely on flags as much as possible (in this case the flag is `None` vs a `list`). This is a good example why, the code would be less error prone and easier to understand if written with two functions, one `cluster_metadata` and the other `topics_metadata`. This would also make it easy to use a caching decorator for this.

With that said I don't think the PR should be blocked because of this, but it would be nice to gradually introduce a safer coding style as we do new changes.